src/PixiEditor/ViewModels/SubViewModels/Document/DocumentViewModel.cs

```diff
@@ -387,18 +387,29 @@ public void MarkAsUnsaved()
     /// </summary>
     /// <param name="includeReference">Should the color be picked from the reference layer</param>
     /// <param name="includeCanvas">Should the color be picked from the canvas</param>
-    public Color PickColor(VecD pos, DocumentScope scope, bool includeReference, bool includeCanvas)
+    /// <param name="referenceTopmost">Is the reference layer topmost. (Only affects the result is includeReference and includeCanvas are set.)</param>
+    public Color PickColor(VecD pos, DocumentScope scope, bool includeReference, bool includeCanvas, bool referenceTopmost = false)
     {
         if (scope == DocumentScope.SingleLayer && includeReference && includeCanvas)
             includeReference = false;
         
         if (includeCanvas && includeReference)
         {
             Color canvasColor = PickColorFromCanvas((VecI)pos, scope);
-            Color? referenceColor = PickColorFromReferenceLayer(pos);
-            if (referenceColor is null)
+            Color? potentialReferenceColor = PickColorFromReferenceLayer(pos);
+            if (potentialReferenceColor is not { } referenceColor)
                 return canvasColor;
-            return ColorHelpers.BlendColors((Color)referenceColor, canvasColor);
+
+            if (!referenceTopmost)
+            {
+                return ColorHelpers.BlendColors(referenceColor, canvasColor);
+            }
+
+            byte referenceAlpha = canvasColor.A == 0 ? referenceColor.A : (byte)(referenceColor.A * 0.6f);
```

Comment:
Would be good to sync 0.6f with like topmost alpha in xaml or wherever it is set

Reply:
wdym?

Reply:
Where did you get 0.6f from?

Reply:
From XAML

Reply:
Exactly, if you change XAML and you forget changing it here, the color will be incorrect